routes/user.py

```diff
@@ -233,3 +266,68 @@ def basic_update_user():
     session.close()
 
     return 'ok'
+
+
+@user_blueprint.route('/user/image', methods=['POST'])
+@roles_required([RoleName.user])
+def upload_image():
+    session = get_session()
+
+    if 'image' not in request.files:
+        return abort(400, 'No file to save')
+
+    image = request.files['image']
+
+    if image.filename == '':
+        return abort(400, 'No files to save')
+
+    ext = image.filename.rsplit('.', 1)[::-1][0]
+
+    if ext not in current_app.config['ALLOWED_EXTENSIONS']:
+        return abort(400, 'This file type is not allowed')
+
+    image.filename = str(uuid4()) + '.' + ext
+    current_path = current_app.config['DIRNAME']
+    assets_path = current_app.config['ASSETS_PATH']
+
+    image.save(os.path.join(current_path, assets_path, image.filename))
+
+    current_user_id = get_jwt_identity()
+    user = session.query(User).get(current_user_id)
+
+    if user.image is not None:
+        path = os.path.join(current_path, assets_path, user.image)
+        if os.path.isfile(path):
+            os.remove(os.path.join(current_path, assets_path, user.image))
+
+    user.image = image.filename
+
+    session.commit()
+    session.close()
+
+    return 'assets/' + image.filename
+
+
+@user_blueprint.route('/user/image', methods=['DELETE'])
+@roles_required([RoleName.user])
+def delete_image():
+    session = get_session()
+
+    current_user_id = get_jwt_identity()
+    user = session.query(User).get(current_user_id)
+
+    current_path = current_app.config['DIRNAME']
+    assets_path = current_app.config['ASSETS_PATH']
+
+    if user.image is not None:
+        path = os.path.join(current_path, assets_path, user.image)
+        if os.path.isfile(path):
+            os.remove(os.path.join(current_path, assets_path, user.image))
+        user.image = None
```

Comment:
I think it's will be better to reset avatar image to gravatar default in this case